### PR TITLE
Fix link/wording for `parse_nt`

### PR DESCRIPTION
**Pull request by JohnTitor**
Follow-up of https://github.com/rust-lang/rust/pull/74826

### CODE REVIEW

File: src/overview.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ we'll talk about that later.
   analysis. The crate entry points for the parser are the `Parser.parse_crate_mod()` and
   `Parser.parse_mod()` methods found in `librustc_parse::parser::item`. The external
   module parsing entry point is `librustc_expand::module::parse_external_mod`. And
-  the macro parser entry point is `rustc_expand::mbe::macro_parser::parse_nt`.
+  the macro parser entry point is [`Parser.parse_nonterminal()`][parse_nonterminal].
```
Comment (mark-i-m) [inline src/overview.md:45]:
```suggestion
  the macro parser entry point is [`Parser::parse_nonterminal()`][parse_nonterminal].
```

Comment (JohnTitor) [reply src/overview.md:45]: I use `.` because we have `Parser.parse_crate_mod()` above. Should we replace them too?

Comment (mark-i-m) [reply src/overview.md:45]: Oh, I didn't notice that. Yeah, would you mind changing the other ones to :: too?

Comment (JohnTitor) [reply src/overview.md]: Sure! Addressed via bbb27ac.

**Pull request merged**